### PR TITLE
Adding make release/new

### PR DESCRIPTION
**Pull request by pseudomuto**
Currently we have one process for patch releases, and a separate one for others. There doesn't seem to be any reason for this, at least not anymore.

With this PR, the new process is simplified to running a single make command, regardless of which type of release you're doing.

Sort of related to #738 

# How it works

Running `make release/new VERSION=<desired-version>` will:

* Validate the version matches `N.N.N(-beta.N)`
* Update the version in version.txt
* Create a new branch named `release-<version>`
* Run `make release/gen-files` to create all the manifests and bundles
* Commit the changes with `Bump version to <version>`

The steps to create the PR and the tag are still the same, and have been outlined in the README for now.

### CODE REVIEW

File: hack/release/steps.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+// GenerateFiles runs make release/gen-files passing the appropriate channel options based on the version.
+func GenerateFiles(fn ExecFn) Step {
+	return StepFn(func(version string) error {
+		ch := "stable"
+		isDefault := "1"
+
+		if strings.Contains(version, "-beta") {
+			ch = "beta"
+			isDefault = "0"
+		}
```
Comment (rail) [inline hack/release/steps.go:64]: ❤️  it!

File: hack/release/main.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+func main() {
+	flag.StringVar(&dir, "dir", ".", "the directory run in")
+	flag.StringVar(&version, "version", "", "the new version to release")
+	flag.Parse()
+
+	steps := []Step{
+		ValidateVersion(),
+		UpdateVersion(),
+		CreateReleaseBranch(process.ExecJUnit),
+		GenerateFiles(process.ExecJUnit),
```
Comment (rail) [inline hack/release/main.go:27]: Maybe also ad a step to make sure we change the version? In other words, that the version in `version.txt` is not the same passed. I have vague memories that the `make release/gen-files` fails if you rerun it...

**Review by rail — APPROVED** [review]
Wooo! Great progress here.

File: hack/release/main.go (hunk 2 above)
Comment (pseudomuto) [reply hack/release/main.go:27]: It does indeed. Nice catch. I'll add a check

File: hack/release/steps.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		return ioutil.WriteFile("version.txt", []byte(version), 0644)
+	})
+}
+
+// CreateReleaseBranch creates a new branch for the release named release-<version>.
+func CreateReleaseBranch(fn ExecFn) Step {
+	return StepFn(func(version string) error {
+		return fn(
+			"git",
+			[]string{"checkout", "-b", fmt.Sprintf("release-%s", version)},
```
[the remark this answers is not in the document]
Comment (pseudomuto) [reply hack/release/steps.go:49]: I think that makes sense. Assuming `origin`_seems_ like a safe bet (I just know I'll be eating my words at some point in the future)

**Comment by pseudomuto** [comment]
Updated to explicitly branch from _origin/master_ and verify that the new version isn't one of the existing git tags.

File: hack/release/main.go (hunk 2 above)
Comment (pseudomuto) [reply hack/release/main.go:47]: I added a check to verify the new version against the existing git tags

File: hack/release/steps.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (pseudomuto) [reply hack/release/steps.go]: It's in there. The trap is set 😄

File: hack/release/main.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+	"sigs.k8s.io/kubetest2/pkg/process"
+)
+
+var (
+	dir     string
+	version string
+
+	deployedBundlesDir = filepath.Join("deploy", "certified-metadata-bundle", "cockroach-operator")
+	versionRegxp       = regexp.MustCompile(`^\d+\.\d+\.\d+(-beta\.\d+)?$`)
```
Comment (davidwding) [inline hack/release/main.go:34]: nit: `deployedBundlesDir` appears unused?
nit: move `versionRegxp` to steps.go where it's used

File: hack/release/steps.go [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+		}
+
+		return nil
+	})
+}
+
+// UpdateVersion sets the version in version.txt
+func UpdateVersion() Step {
+	return StepFn(func(version string) error {
+		return ioutil.WriteFile("version.txt", []byte(version), 0644)
```
Comment (davidwding) [inline hack/release/steps.go:88]: can you add a comment on what 0644 means? for those of us who aren't totally fluent in chmod values :)

File: hack/release/steps.go [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+	})
+}
+
+// EnsureUniqueVersion verifies that this is a new version by checking the existing tags.
+func EnsureUniqueVersion(fn CmdFn) Step {
+	return StepFn(func(version string) error {
+		stdout := new(bytes.Buffer)
+		stderr := new(bytes.Buffer)
+
+		cmd := exec.Command("git", "tag")
```
Comment (davidwding) [inline hack/release/steps.go:63]: think it's worth using some git api like `go-git` or is that overkill?

File: hack/release/main.go (hunk 4 above)
Comment (pseudomuto) [reply hack/release/main.go]: Nice catch! Before checking the tags, I was initially going to search the deploy directory. Removed.

File: hack/release/steps.go (hunk 5 above)
Comment (pseudomuto) [reply hack/release/steps.go:94]: Done

**Review by davidwding — APPROVED** [review]

**Pull request merged**